### DISCUSSION TITLE
Huggingface Inference API

### DISCUSSION THREAD

**Discussion by alberto98fx** [open-webui/open-webui#4122] [General]
It would be cool to allow users to use HF inference API from the WebUI; they do not seem to support OpenAI libraries hence it's not possible to use the existing connector.
For Pro users of HF using inference means higher rate limits for just $9 per month! 

A possible way out of this is LiteLLM but since it has been removed from WebUI it is a bit more complex to setup 

https://huggingface.co/inference-api/serverless

**Comment by tjbck** [comment]
Contributions welcome here! https://openwebui.com/functions

**Comment by tazztone** [comment]
well, some seem to be working: 
![image](https://github.com/user-attachments/assets/3e310a53-fa7c-4ecf-97bf-78a70fd262a1)
adding the https://huggingface.co/api/models will populate the model list with a ton of models. this llama among them. however i miss qwen. maybe the list is just too long to incorporate all models into the open-webui dropdown
EDIT: sorry this llama-3.2 is from qroq released yesterday

**Comment by artemiux** [comment]
Try [this](https://openwebui.com/f/artemium/hugging_face_inference_api_proxy). Hardcode your models into `pipes()` to be shown in Open WebUI.
